hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarbonitedMetadataThawingActor.scala

```diff
@@ -100,34 +110,47 @@ object CarbonitedMetadataThawingActor {
-          case MetadataQuery(_, Some(_), Some(_), None, None, _) => json
-          case MetadataQuery(_, Some(_), None, _, _, _) => json
-          case wrong => throw new RuntimeException(s"Programmer Error: Invalid MetadataQuery: $wrong")
-        }
-        // For carbonited metadata, "expanded" subworkflows translates to not deleting subworkflows out of the root workflow that already
-        // contains them. So `_.validNel` for expanded subworkflows and `JsonEditor.removeSubworkflowMetadata` for unexpanded subworkflows.
-        val processSubworkflowMetadata: Json => ErrorOr[Json] = if (get.key.expandSubWorkflows) _.validNel else JsonEditor.replaceSubworkflowMetadataWithId
-        processSubworkflowMetadata(intermediate)
-      case other =>
-        throw new RuntimeException(s"Programmer Error: Unexpected BuildWorkflowMetadataJsonAction message of type '${other.getClass.getSimpleName}': $other")
+    def editFor(action: BuildWorkflowMetadataJsonAction, isRootWorkflow: Boolean): ErrorOr[Json] = {
+      val rootOrSubWorkflowJson = if (isRootWorkflow) json.validNel else json.extractSubworkflowMetadata(action.workflowId.toString)
+      action match {
+        case _: GetLogs => rootOrSubWorkflowJson.map(JsonEditor.logs)
+        case _: WorkflowOutputs => rootOrSubWorkflowJson.map(JsonEditor.outputs)
+        case get: GetMetadataAction =>
+          val intermediate = get.key match {
+            case MetadataQuery(_, None, None, None, None, _) => rootOrSubWorkflowJson
+            case MetadataQuery(_, None, Some(key), None, None, _) => rootOrSubWorkflowJson.map(JsonEditor.includeJson(_, NonEmptyList.of(key)))
+            case MetadataQuery(_, None, None, includeKeys, excludeKeys, _) => rootOrSubWorkflowJson.map(JsonEditor.includeExcludeJson(_, includeKeys, excludeKeys))
+            // The following three don't bother with exclusions or inclusions since they are only used internally
+            // and the existing client code is robust to superfluous data.
+            case MetadataQuery(_, Some(_), None, None, None, _) => rootOrSubWorkflowJson
+            case MetadataQuery(_, Some(_), Some(_), None, None, _) => rootOrSubWorkflowJson
+            case MetadataQuery(_, Some(_), None, _, _, _) => rootOrSubWorkflowJson
+            case wrong => throw new RuntimeException(s"Programmer Error: Invalid MetadataQuery: $wrong")
+          }
+          // For carbonited metadata, "expanded" subworkflows translates to not deleting subworkflows out of the root workflow that already
+          // contains them. So `inpJson` for expanded subworkflows and `JsonEditor.replaceSubworkflowMetadataWithId` for unexpanded subworkflows.
+          val processSubworkflowMetadata: ErrorOr[Json] => ErrorOr[Json] = {
+            inpJson => if (get.key.expandSubWorkflows) inpJson else inpJson.flatMap(JsonEditor.replaceSubworkflowMetadataWithId)
+          }
+          processSubworkflowMetadata(intermediate)
```

Comment:
This section seems a little weird to me... why not replace the "function as a val followed by exactly one usage" with just an inline evaluation? eg:
```
if (get.key.expandSubWorkflows) intermediate else intermediate.flatMap(JsonEditor.replaceSubworkflowMetadataWithId)
```

Reply:
I guess it was introduced before to make code more clear, but given that there's also a comment, function val seems redundant indeed.